src/ucp/wireup/wireup_cm.c

```diff
@@ -733,21 +737,31 @@ static void ucp_cm_server_connect_cb(uct_ep_h ep, void *arg,
     uct_worker_cb_id_t prog_id = UCS_CALLBACKQ_ID_NULL;
     ucp_lane_index_t cm_lane;
 
-    if (status != UCS_OK) {
+    if (status == UCS_OK) {
+        uct_worker_progress_register_safe(ucp_ep->worker->uct,
+                                          ucp_cm_server_connect_progress,
+                                          ucp_ep, UCS_CALLBACKQ_FLAG_ONESHOT,
+                                          &prog_id);
+        ucp_worker_signal_internal(ucp_ep->worker);
+    } else if (status == UCS_ERR_CONNECTION_RESET) {
+        /* remote side initiated disconnect before local side has completed
+         * connection establishment, so:
+         * 1) establish connection to complete any pending requests from wireup
+         *    lane
+         * 2) handle disconnect same way as close protocol */
+        uct_worker_progress_register_safe(ucp_ep->worker->uct,
+                                          ucp_cm_server_connect_progress,
```

Comment:
why do we need to invoke ucp_cm_server_connect_progress, if we are already disconnecting? we don't want to let transports send things, instead need to move ep to error state

Reply:
To convert wireup ep to real one and handle pending queue. Moving ep to failed state may fast-forward flush request scheduled by `ucp_ep_close_nb` for example.
We may need to rework this part as well during implementing another failed EP state which would block sending after disconnect event. But IMO this is important to have stable tests+UCX to continue with close protocol enhancements.

Reply:
If no easy fix now pls add TODO

Reply:
done https://github.com/openucx/ucx/pull/4761/commits/da970e41c6ad4e0c0a3f6a902ca9ed45eddef3d1